jvm-packages/xgboost4j-spark-gpu/src/test/scala/ml/dmlc/xgboost4j/scala/spark/GpuXGBoostPluginSuite.scala

```diff
@@ -357,25 +393,101 @@ class GpuXGBoostPluginSuite extends GpuTestSuite {
     }
   }
 
-  Seq("binary:logistic", "multi:softprob").foreach { case objective =>
-    test(s"$objective: XGBoost-Spark should match xgboost4j") {
-      withGpuSparkSession() { spark =>
-        import spark.implicits._
+  Seq(false, true).foreach { useExtMem =>
+    Seq("binary:logistic", "multi:softprob").foreach { objective =>
+      test(s"$objective: XGBoost-Spark should match xgboost4j with useExtMem: ${useExtMem}") {
+        withGpuSparkSession() { spark =>
+          import spark.implicits._
 
-        val numRound = 100
-        var xgboostParams: Map[String, Any] = Map(
-          "objective" -> objective,
-          "device" -> "cuda"
-        )
+          val numRound = 100
+          var xgboostParams: Map[String, Any] = Map(
+            "objective" -> objective,
+            "device" -> "cuda"
+          )
+
+          val (trainPath, testPath) = if (objective == "binary:logistic") {
+            (writeFile(Classification.train.toDF("label", "weight", "c1", "c2", "c3")),
+              writeFile(Classification.test.toDF("label", "weight", "c1", "c2", "c3")))
+          } else {
+            xgboostParams = xgboostParams ++ Map("num_class" -> 6)
+            (writeFile(MultiClassification.train.toDF("label", "weight", "c1", "c2", "c3")),
+              writeFile(MultiClassification.test.toDF("label", "weight", "c1", "c2", "c3")))
+          }
+
+          val df = spark.read.parquet(trainPath)
+          val testdf = spark.read.parquet(testPath)
+
+          val features = Array("c1", "c2", "c3")
+          val featuresIndices = features.map(df.schema.fieldIndex)
+          val label = "label"
+
+          val classifier = new XGBoostClassifier(xgboostParams)
+            .setFeaturesCol(features)
+            .setLabelCol(label)
+            .setNumRound(numRound)
+            .setLeafPredictionCol("leaf")
+            .setContribPredictionCol("contrib")
+            .setDevice("cuda")
```

Comment:
Can we have an error test that shows when CPU is used there's a clear error message?

Reply:
I added the test.